diesel/src/mysql/connection/raw.rs

```diff
@@ -0,0 +1,97 @@
+extern crate mysqlclient_sys as ffi;
+
+use std::ffi::CStr;
+use std::os::{raw as libc};
+use std::ptr;
+use std::sync::{Once, ONCE_INIT};
+
+use result::{ConnectionResult, ConnectionError};
+use super::url::ConnectionOptions;
+
+pub struct RawConnection(*mut ffi::MYSQL);
+
+impl RawConnection {
+    pub fn new() -> Self {
+        perform_thread_unsafe_library_initialization();
+        let raw_connection = unsafe { ffi::mysql_init(ptr::null_mut()) };
+        if raw_connection.is_null() {
+            panic!("Insufficient memory to allocate connection");
```

Comment:
I'd add a comment that this is the error case that https://dev.mysql.com/doc/refman/5.7/en/mysql-init.html documents, trusting it's actually the only case where NULL is returned.